packages/react-form-renderer/src/hooks/use-field-provider-api.js

```diff
@@ -0,0 +1,59 @@
+import { useEffect } from 'react';
+import { useField } from 'react-final-form';
+import useFormApi from './useFormApi';
+import enhancedOnChange from '../form-renderer/enhanced-on-change';
+
+const useFieldProviderApi = ({ name, initializeOnMount, component, render, ...props }) => {
+  const formOptions = useFormApi();
+  const fieldProps = useField(name, props);
+
+  useEffect(() => {
+    /**
+     * Re initialize field when mounted to the Form
+     * This affects conditional fields
+     */
+    if (initializeOnMount) {
+      const initialValue = props.initialValue || fieldProps.meta.initial;
+      fieldProps.input.onChange(initialValue);
+    }
+
+    return () => {
+      /**
+       * Delete the value from form state when field is inmounted
+       */
+      if ((formOptions.clearOnUnmount || props.clearOnUnmount) && props.clearOnUnmount !== false) {
+        fieldProps.input.onChange(undefined);
+      }
+    };
+  }, []);
+
+  /**
+   * Prepare deleted value of field
+   */
+  const fieldClearedValue = props.hasOwnProperty('clearedValue') ? props.clearedValue : formOptions.clearedValue;
+
+  /**
+   * construct component props necessary that would live in field provider
+   */
+  return {
+    ...props,
+    formOptions,
```

Comment:
Do we really want to pass formOptions here? I thought that was the reason we made the separate hook.

Reply:
I want to make props clean up in the next step. There is a lot more stuff like that.

Reply:
I have already cleaned up formOptions in the PR with that hook, so I am think it should be ready to be removed now.